src/mark.c

```diff
@@ -170,13 +182,24 @@ gboolean r_mark_good(RaucSlot *slot, GError **error)
 	g_return_val_if_fail(slot, FALSE);
 	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
 
+	/* It is more sensible to mark the slot as good before locking the counter. As in case of a power fail between
```

Comment:
Above, this section is attached to the `R_CONFIG_FALLBACK_LOCK_COUNTER` check, which also seems more appropriate here since for 90% of the cases, the `r_boot_set_state()` will not have anything to do with slot locking.